sos/utilities.py

```diff
@@ -301,19 +320,25 @@ def _check_poller(proc):
                 except Exception:
                     p.terminate()
                     if to_file:
-                        _output.close()
+                        if tac:
+                            with open(to_file, 'wb') as f_dst:
+                                tac_logs(_output, f_dst, True)
                     # until we separate timeouts from the `timeout` command
                     # handle per-cmd timeouts via Plugin status checks
                     reader.running = False
                     return {'status': 124, 'output': reader.get_contents(),
                             'truncated': reader.is_full}
-            if to_file:
-                _output.close()
 
             # wait for Popen to set the returncode
             while p.poll() is None:
                 pass
 
+            if to_file:
+                if tac:
```

Comment:
Also nitpick now: why not merging into single `if to_file and tac:`?

Reply:
There was something else in the `if to_file:` at some point (will fix on the next push)

Reply:
Fixed